test/integ/test_cursor.py

```diff
@@ -1039,11 +1042,16 @@ def test_empty_execution(conn, sql):
     "reuse_results", (False, pytest.param(True, marks=pytest.mark.skipolddriver))
 )
 def test_reset_fetch(conn, reuse_results):
-    """Tests behavior after resetting the cursor."""
+    """Tests behavior after resetting an open cursor."""
     with conn(reuse_results=reuse_results) as cnx:
         with cnx.cursor() as cur:
             cur.execute("select 1")
+            assert cur.rowcount == 1
             cur.reset()
+            assert (
+                cur.rowcount is None
```

Comment:
`rowcount` should never be None, it should be `-1` when it is reset according to https://peps.python.org/pep-0249/#rowcount

Reply:
The internal `_total_rowcount` behind this attribute method is `-1` but the property method wraps and returns `None` in that case.

This test was just asserting a behavior that has been the same since 2017 after this commit (I could not find a reason for this change): https://github.com/snowflakedb/snowflake-connector-python/commit/534674ad46d121a2249790814b06cac8876ce217#diff-6ef89a5da3fcd97c38c8bf752b68deb48e573ea5557334e5a5dd9d89ddcad03e

Perhaps the change above was made following the note in the PEP-249?

> Future versions of the DB API specification could redefine the latter case to have the object return None instead of -1.

If the behavior of the property method `cursor.rowcount` needs to be changed back to returning `-1` we can perhaps do it over a separate JIRA?